iris-mpc-gpu/tests/e2e.rs

```diff
@@ -249,13 +249,16 @@ mod e2e_test {
 
             for idx in 0..batch_size {
                 let request_id = Uuid::new_v4();
+                let mut use_or_rule_for_serial_ids: Vec<u32> = vec![];
                 // Automatic random tests
                 let options = if responses.is_empty() {
                     3
                 } else if deleted_indices_buffer.is_empty() {
                     4
-                } else {
+                } else if rng.gen() {
```

Comment:
is there a reason why this is only taken 50% of the time?

Reply:
but this is not doing the actual generation, this is just the available options for the rng in line 277. I think it is just fine to have this always be an option. The only reason this is as convoluted as it is here, is that we cannot do some of the options unless certain events happened on previous queries (e.g, deletion, insertion)

Reply:
Probably this should be refactored to just be a more descriptive enum + random.choice

Reply:
Makes sense, I'll think about refactoring this in a separate PR. It was indeed a bit convoluted

Reply:
can also do that in a follow up PR